Lib/test/test_format.py

```diff
@@ -519,5 +519,67 @@ def test_with_an_underscore_and_a_comma_in_format_specifier(self):
         with self.assertRaisesRegex(ValueError, error_msg):
             '{:_,}'.format(1)
 
+    def test_better_error_message_format(self):
+        # https://bugs.python.org/issue20524
+        complex_err = re.escape(
+            "Invalid format specifier '%M' for object of type 'complex'")
+        with self.assertRaisesRegex(ValueError, complex_err):
+            "{l:%M}".format(l=complex(12))
+
+        int_err = re.escape(
+            "Invalid format specifier '%M:%S' "
+            "for object of type 'int'")
+        with self.assertRaisesRegex(ValueError, int_err):
+            "{length:%M:%S}".format(length=12)
+
+        float_err = re.escape(
+            "Invalid format specifier '%S' for object of type 'float'")
+        with self.assertRaisesRegex(ValueError, float_err):
+            "{0:%S}".format(12.5)
+
+        str_err = re.escape(
+            "Invalid format specifier '%%k' for object of type 'str'")
+        with self.assertRaisesRegex(ValueError, str_err):
+            "{a:%%k}".format(a='a')
+
+    def test_unicode_in_error_message(self):
```

Comment:
I hadn't noticed this test before. It's great, especially because of the work in using "kind" to compute the actual error message.